File: examples/typescript/01_node_info.ts
// Scenarios 1-3: connect to a local node; show blockchain and network info.
// Run: npx tsx examples/typescript/01_node_info.ts  (with KINGPEPE_RPC_* env vars)
import { KingPepeClient } from "@kingpepe/sdk";

async function main() {
  const client = KingPepeClient.fromEnv();

  if (!(await client.healthCheck())) {
    throw new Error("KingPepe node is not reachable — check KINGPEPE_RPC_* env vars.");
  }

  const chain = await client.getBlockchainInfo();
  console.log("Blockchain:");
  console.log(`  chain=${chain.chain} height=${chain.blocks} headers=${chain.headers}`);
  console.log(`  bestblockhash=${chain.bestblockhash}`);
  console.log(`  ibd=${chain.initialblockdownload} pruned=${chain.pruned}`);

  const net = await client.getNetworkInfo();
  console.log("Network:");
  console.log(`  subversion=${net.subversion} protocol=${net.protocolversion}`);
  console.log(`  connections=${net.connections} active=${net.networkactive}`);
}

main().catch((err) => {
  console.error("error:", err.message);
  process.exit(1);
});

File: examples/python/01_node_info.py
"""Scenarios 1-3: connect to a local node; show blockchain and network info.

Run: python examples/python/01_node_info.py   (with KINGPEPE_RPC_* env vars)
"""

from __future__ import annotations

import sys

from kingpepe_sdk import KingPepeClient


def main() -> int:
    client = KingPepeClient.from_env()
    if not client.health_check():
        print("KingPepe node is not reachable — check KINGPEPE_RPC_* env vars.", file=sys.stderr)
        return 1

    chain = client.get_blockchain_info()
    print("Blockchain:")
    print(f"  chain={chain['chain']} height={chain['blocks']} headers={chain['headers']}")
    print(f"  bestblockhash={chain['bestblockhash']}")

    net = client.get_network_info()
    print("Network:")
    print(f"  subversion={net['subversion']} protocol={net['protocolversion']}")
    print(f"  connections={net['connections']} active={net['networkactive']}")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
